### PR TITLE
fix(ssh): reimplement ssh port

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description
ssh_port has been removed when replacing centcore by gorgone.

But ssh_port is used by service discovery to reach Remote Servers and Pollers.

This Pull Request take ssh_port back

**Fixes** MON-5110

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

Go to Configuration > Pollers > Add (or Edit) . You should see a new field "SSH Legacy Port" in the form (but not on a Central Poller)

## Checklist

- [x] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: doc/fr/api/clapi/objects/instances.rst [hunk 1]
```diff
@@ -129,6 +129,8 @@ nagios_bin                 Path of the Scheduler binary
 
 nagiostats_bin             Path of the Nagios Stats binary
 
+ssh_port                   SSH Port
+
```
Comment (sc979) [inline doc/fr/api/clapi/objects/instances.rst:133]: shouldn't the SSH port option, be after the gorgone_communication_type ?

Comment (cgagnaire) [reply doc/fr/api/clapi/objects/instances.rst:133]: Actually it's the legacy port used until Gorgone, but nor by Gorgone, so different from gorgone_port, so before gorgone_communication_type .

Comment (sc979) [reply doc/fr/api/clapi/objects/instances.rst:133]: but in gorgone_communication_type you need to specify if you use the ZMQ or the SSH ;)

Comment (cgagnaire) [reply doc/fr/api/clapi/objects/instances.rst:133]: I know but SSH communication type also refer to gorgone_port.
ssh_port is not used by Gorgone.

File: www/include/configuration/configServers/DB-Func.php [hunk 2]
```diff
@@ -501,6 +501,12 @@ function insertServer(array $data): int
     } else {
         $rq .= "NULL, ";
     }
+    if (isset($data["ssh_port"]) && $data["ssh_port"] != null) {
```
Comment (sc979) [inline www/include/configuration/configServers/DB-Func.php:504]: you can use !empty instead

File: www/include/configuration/configServers/DB-Func.php [hunk 3]
```diff
@@ -776,6 +782,13 @@ function updateServer(int $id, array $data): void
     } else {
         $rq .= "NULL, ";
     }
+    $rq .= "`ssh_port` = ";
+    if (isset($data["ssh_port"]) && $data["ssh_port"] != null) {
```
Comment (sc979) [inline www/include/configuration/configServers/DB-Func.php:786]: you can use !empty instead

File: www/class/centreon-clapi/centreonInstance.class.php [hunk 4]
```diff
@@ -108,15 +110,21 @@ public function initInsertParameters($parameters)
         $addParams[$this->object->getUniqueLabelField()] = $params[self::ORDER_UNIQUENAME];
         $addParams['ns_ip_address'] = $params[self::ORDER_ADDRESS];
 
+        if(is_numeric($params[self::ORDER_GORGONE_PROTOCOL])){
+           $revertGorgoneCom = array_flip (self::GORGONE_COMMUNICATION);
```
Comment (loiclau) [inline www/class/centreon-clapi/centreonInstance.class.php:114]:
```suggestion
            $revertGorgoneCom = array_flip (self::GORGONE_COMMUNICATION);
```

**Pull request merged**